src/fragments/start/getting-started/reactnative/auth.mdx

```diff
@@ -0,0 +1,378 @@
+  return (
+    <Pressable onPress={signOut} style={styles.buttonContainer}>
+      <Text style={styles.buttonText}>
+        Hello, {user.username}! Click here to sign out!
+      </Text>
+    </Pressable>
+  );
+};
+
+const initialFormState = {name: '', description: ''};
+
+const App = () => {
+  const [formState, setFormState] = useState(initialFormState);
+  const [todos, setTodos] = useState([]);
+
+  useEffect(() => {
+    fetchTodos();
+  }, []);
+
+  function setInput(key, value) {
+    setFormState({...formState, [key]: value});
+  }
+
+  async function fetchTodos() {
+    try {
+      const todoData = await API.graphql(graphqlOperation(listTodos));
+      const todos = todoData.data.listTodos.items;
+      setTodos(todos);
+    } catch (err) {
+      console.log('error fetching todos');
+    }
+  }
+
+  async function addTodo() {
+    try {
+      if (!formState.name || !formState.description) return;
+      const todo = {...formState};
+      setTodos([...todos, todo]);
+      setFormState(initialFormState);
+      await API.graphql(graphqlOperation(createTodo, {input: todo}));
+    } catch (err) {
+      console.log('error creating todo:', err);
+    }
+  }
+
+  return (
+    <SafeAreaView style={styles.container}>
+      <View style={styles.container}>
+        <SignOutButton />
+        <TextInput
+          onChangeText={value => setInput('name', value)}
+          style={styles.input}
+          value={formState.name}
+          placeholder="Name"
+        />
+        <TextInput
+          onChangeText={value => setInput('description', value)}
+          style={styles.input}
+          value={formState.description}
+          placeholder="Description"
+        />
+        <Pressable onPress={addTodo} style={styles.buttonContainer}>
+          <Text style={styles.buttonText}>Create todo</Text>
+        </Pressable>
+        {todos.map((todo, index) => (
+          <View key={todo.id ? todo.id : index} style={styles.todo}>
+            <Text style={styles.todoName}>{todo.name}</Text>
+            <Text style={styles.todoDescription}>{todo.description}</Text>
+          </View>
+        ))}
+      </View>
+    </SafeAreaView>
+  );
+};
+
+export default withAuthenticator(App);
+
+const styles = StyleSheet.create({
+  container: {width: 400, flex: 1, padding: 20, alignSelf: 'center'},
+  todo: {marginBottom: 15},
+  input: {backgroundColor: '#ddd', marginBottom: 10, padding: 8, fontSize: 18},
+  todoName: {fontSize: 20, fontWeight: 'bold'},
+  buttonContainer: {
+    alignSelf: 'center',
+    backgroundColor: 'black',
+    paddingHorizontal: 8,
+  },
+  buttonText: {color: 'white', padding: 16, fontSize: 18},
+});
+```
+
+</Block>
+</BlockSwitcher>
+
+
+Using Amplify UI connected components makes it easier to manage styling across your entire app.
+
+In this example, you used the Amplify React UI library and the `withAuthenticator` Higher-Order Component to quickly get up and running with a real-world authentication flow.
```

Comment:
the use of "Higher-order" component here is a little weird to me. Aren't we calling them "connected components"?

Reply:
understood. resolving this!